tests/sentry/flags/providers/test_launchdarkly.py

```diff
@@ -0,0 +1,402 @@
+                "production": {
+                    "on": False,
+                    "archived": False,
+                    "salt": "ee006f821c8443538c49541256188073",
+                    "sel": "d928ffd4789949d3b8956a35e3ded70f",
+                    "lastModified": 1729123465190,
+                    "version": 1,
+                    "targets": [],
+                    "contextTargets": [],
+                    "rules": [],
+                    "fallthrough": {"variation": 0},
+                    "offVariation": 1,
+                    "prerequisites": [],
+                    "_site": {
+                        "href": "/default/production/features/test-flag",
+                        "type": "text/html",
+                    },
+                    "_environmentName": "Production",
+                    "trackEvents": False,
+                    "trackEventsFallthrough": False,
+                },
+            },
+        },
+    }
+
+    res = handle_provider_event("launchdarkly", request_data, "123")
+    assert len(res) == 1
+    flag_row = res[0]
+    assert flag_row["action"] == ACTION_MAP["updated"]
+
+
+def test_launchdarkly_delete():
+    request_data = {
+        "_id": "1234",
+        "_accountId": "1234",
+        "date": 1729123867537,
+        "accesses": [
+            {"action": "deleteFlag", "resource": "proj/default:env/test:flag/test-flag"},
+        ],
+        "kind": "flag",
+        "name": "test flag",
+        "description": "deleted the flag",
+        "shortDescription": "",
+        "comment": "",
+        "member": {
+            "_links": {
+                "parent": {"href": "/api/v2/members", "type": "application/json"},
+                "self": {
+                    "href": "/api/v2/members/1234",
+                    "type": "application/json",
+                },
+            },
+            "_id": "1234",
+            "email": "michelle@example.io",
+            "firstName": "Michelle",
+            "lastName": "Doe",
+        },
+        "titleVerb": "deleted the flag",
+        "title": "Michelle deleted the flag [test flag](https://app.launchdarkly.com/default/test/features/test-flag) in `Test`",
+    }
+
+    res = handle_provider_event("launchdarkly", request_data, "123")
+    assert len(res) == 1
+    flag_row = res[0]
+    assert flag_row["action"] == ACTION_MAP["deleted"]
+
+
+def test_bad_launchdarkly_data():
+    request_data = {
+        "accesses": [],
+        "member": {
+            "_links": {
+                "parent": {"href": "/api/v2/members", "type": "application/json"},
+                "self": {
+                    "href": "/api/v2/members/1234",
+                    "type": "application/json",
+                },
+            },
+            "_id": "1234",
+            "firstName": "Michelle",
+            "lastName": "Doe",
+        },
+    }
+    with pytest.raises(DeserializationError):
+        handle_provider_event("launchdarkly", request_data, "123")
```

Comment:
Its helpful to assert what failed to guard against regressions. `assert "Missing first_name"` in errors or something.